dotnet/src/Connectors/Connectors.AzureAIInference/Extensions/AzureAIInferenceServiceCollectionExtensions.cs

```diff
@@ -18,71 +19,107 @@ namespace Microsoft.SemanticKernel;
 public static class AzureAIInferenceServiceCollectionExtensions
 {
     /// <summary>
-    /// Adds the <see cref="AzureAIInferenceChatCompletionService"/> to the <see cref="IServiceCollection"/>.
+    /// Adds a Azure AI Inference <see cref="IChatCompletionService"/> to the <see cref="IServiceCollection"/>.
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection"/> instance to augment.</param>
-    /// <param name="modelId">Target Model Id for endpoints supporting more than one model</param>
+    /// <param name="modelId">Target Model Id</param>
     /// <param name="apiKey">API Key</param>
     /// <param name="endpoint">Endpoint / Target URI</param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="serviceId">A local identifier for the given AI service</param>
     /// <returns>The same instance as <paramref name="services"/>.</returns>
     public static IServiceCollection AddAzureAIInferenceChatCompletion(
         this IServiceCollection services,
-        string? modelId = null,
+        string modelId,
         string? apiKey = null,
         Uri? endpoint = null,
         HttpClient? httpClient = null,
         string? serviceId = null)
     {
         Verify.NotNull(services);
 
-        AzureAIInferenceChatCompletionService Factory(IServiceProvider serviceProvider, object? _) =>
-            new(modelId,
-                apiKey,
-                endpoint,
-                HttpClientProvider.GetHttpClient(httpClient, serviceProvider),
-                serviceProvider.GetService<ILoggerFactory>());
+        services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
+        {
+            var chatClientBuilder = new ChatClientBuilder()
+               .UseFunctionInvocation(config =>
```

Comment:
We may want to switch in our own function calling middleware which supports our function calling abstractions and filters. Let's discuss this post Ignite.